Enable tests disabled in #154

Tests FindNullDerefIntegrationTest and FindUnsatisfiedObligationTest were disabled in #154 (for #151) because they were failing due the issues with 
the new AnalysisRunner which seem to be confused by an unexpected findbugs.xml location:
`
java.lang.ExceptionInInitializerError
	at edu.umd.cs.findbugs.test.SpotBugsRule.before(SpotBugsRule.java:52)
	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:46)
	at org.junit.rules.RunRules.evaluate(RunRules.java:20)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.gradle.api.internal.tasks.testing.junit.JUnitTestClassExecuter.runTestClass(JUnitTestClassExecuter.java:114)
	at org.gradle.api.internal.tasks.testing.junit.JUnitTestClassExecuter.execute(JUnitTestClassExecuter.java:57)
	at org.gradle.api.internal.tasks.testing.junit.JUnitTestClassProcessor.processTestClass(JUnitTestClassProcessor.java:66)
	at org.gradle.api.internal.tasks.testing.SuiteTestClassProcessor.processTestClass(SuiteTestClassProcessor.java:51)
	at sun.reflect.GeneratedMethodAccessor5.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:35)
	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:24)
	at org.gradle.internal.dispatch.ContextClassLoaderDispatch.dispatch(ContextClassLoaderDispatch.java:32)
	at org.gradle.internal.dispatch.ProxyDispatchAdapter$DispatchingInvocationHandler.invoke(ProxyDispatchAdapter.java:93)
	at com.sun.proxy.$Proxy2.processTestClass(Unknown Source)
	at org.gradle.api.internal.tasks.testing.worker.TestWorker.processTestClass(TestWorker.java:109)
	at sun.reflect.GeneratedMethodAccessor4.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:35)
	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:24)
	at org.gradle.internal.remote.internal.hub.MessageHub$Handler.run(MessageHub.java:377)
	at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:54)
	at org.gradle.internal.concurrent.StoppableExecutorImpl$1.run(StoppableExecutorImpl.java:40)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
Caused by: java.nio.file.FileSystemNotFoundException
	at com.sun.nio.zipfs.ZipFileSystemProvider.getFileSystem(ZipFileSystemProvider.java:171)
	at com.sun.nio.zipfs.ZipFileSystemProvider.getPath(ZipFileSystemProvider.java:157)
	at java.nio.file.Paths.get(Paths.java:143)
	at edu.umd.cs.findbugs.test.AnalysisRunner.createTempJar(AnalysisRunner.java:146)
	at edu.umd.cs.findbugs.test.AnalysisRunner.<clinit>(AnalysisRunner.java:61)
	... 36 more
`

--- a/spotbugs-tests/src/test/java/edu/umd/cs/findbugs/detect/FindUnsatisfiedObligationTest.java
+++ b/spotbugs-tests/src/test/java/edu/umd/cs/findbugs/detect/FindUnsatisfiedObligationTest.java
@@ -6,12 +6,10 @@ import static org.junit.Assert.assertThat;
 
 import java.nio.file.Paths;
 
-import org.junit.Ignore;
 import org.junit.Rule;
 import org.junit.Test;
 
 import edu.umd.cs.findbugs.BugCollection;
-import edu.umd.cs.findbugs.annotations.DischargesObligation;
 import edu.umd.cs.findbugs.test.SpotBugsRule;
 
 public class FindUnsatisfiedObligationTest {
@@ -23,7 +21,6 @@ public class FindUnsatisfiedObligationTest {
      *      issue</a>
      */
     @Test
-    @Ignore
     public void testIssue60() {
         BugCollection bugCollection = spotbugs.performAnalysis(Paths.get("../spotbugsTestCases/build/classes/main/Issue60.class"));
         assertThat(bugCollection, is(emptyIterable()));

--- a/spotbugs-tests/src/test/java/edu/umd/cs/findbugs/integration/FindNullDerefIntegrationTest.java
+++ b/spotbugs-tests/src/test/java/edu/umd/cs/findbugs/integration/FindNullDerefIntegrationTest.java
@@ -24,7 +24,6 @@ import static org.junit.Assert.assertThat;
 
 import java.io.IOException;
 
-import org.junit.Ignore;
 import org.junit.Test;
 
 import edu.umd.cs.findbugs.AbstractIntegrationTest;
@@ -34,7 +33,6 @@ import edu.umd.cs.findbugs.test.matcher.BugInstanceMatcherBuilder;
 public class FindNullDerefIntegrationTest extends AbstractIntegrationTest {
 
     @Test
-    @Ignore
     public void testNullFromReturnOnLambda() {
         performAnalysis("Elvis.class");
 
@@ -53,7 +51,6 @@ public class FindNullDerefIntegrationTest extends AbstractIntegrationTest {
     }
 
     @Test
-    @Ignore
     public void testLambdaIssue20() throws IOException, InterruptedException {
         performAnalysis("lambdas/Issue20.class");
 

--- a/test-harness/src/main/java/edu/umd/cs/findbugs/test/AnalysisRunner.java
+++ b/test-harness/src/main/java/edu/umd/cs/findbugs/test/AnalysisRunner.java
@@ -6,7 +6,10 @@ import java.io.File;
 import java.io.IOException;
 import java.io.InputStream;
 import java.io.OutputStream;
+import java.net.JarURLConnection;
+import java.net.URI;
 import java.net.URISyntaxException;
+import java.net.URL;
 import java.nio.file.FileVisitResult;
 import java.nio.file.Files;
 import java.nio.file.Path;
@@ -41,7 +44,7 @@ import edu.umd.cs.findbugs.plugins.DuplicatePluginIdException;
  * auxClasspathEntries should be specified before you invoke {@link #run(Path...)}
  * method.
  * </p>
- * 
+ *
  * @since 3.1
  */
 @ParametersAreNonnullByDefault
@@ -133,17 +136,27 @@ public class AnalysisRunner {
      * Create a jar file which contains all resource files. This is necessary to
      * let {@link Plugin#loadCustomPlugin(File, Project)} load custom plugin to
      * test.
-     * 
+     *
      * @return a {@link File} instance which represent generated jar file
      * @throws IOException
      * @throws URISyntaxException
      */
     private static File createTempJar() throws IOException, URISyntaxException {
         ClassLoader cl = AnalysisRunner.class.getClassLoader();
+
+        URL resource = cl.getResource("findbugs.xml");
+        URI uri = resource.toURI();
+
+        if ("jar".equals(uri.getScheme())) {
+            JarURLConnection connection = (JarURLConnection) resource.openConnection();
+            URL url = connection.getJarFileURL();
+            return new File(url.getFile());
+        }
+
         Path tempJar = File.createTempFile("SpotBugsAnalysisRunner", ".jar").toPath();
         try (OutputStream output = Files.newOutputStream(tempJar, StandardOpenOption.WRITE);
                 JarOutputStream jar = new JarOutputStream(output)) {
-            Path resourceRoot = Paths.get(cl.getResource("findbugs.xml").toURI()).getParent();
+            Path resourceRoot = Paths.get(uri).getParent();
 
             byte[] data = new byte[4 * 1024];
             Files.walkFileTree(resourceRoot, new SimpleFileVisitor<Path>() {

